src/main/kotlin/org/rust/ide/refactoring/changeSignature/RsChangeSignatureDialog.kt

```diff
@@ -356,16 +371,30 @@ private class ChangeSignatureDialog(project: Project, descriptor: SignatureDescr
 private fun createTypeCodeFragment(
     context: RsElement,
     type: RsTypeReference?
+): PsiCodeFragment = createCodeFragment(context) { importTarget ->
+    RsTypeReferenceCodeFragment(
+        context.project,
+        type?.text.orEmpty(),
+        context = importTarget,
+        importTarget = importTarget
+    )
+}
+private fun createExprCodeFragment(context: RsElement): PsiCodeFragment = createCodeFragment(context) { importTarget ->
+    RsExpressionCodeFragment(
+        context.project,
+        "",
+        context = importTarget,
+        importTarget = importTarget
+    )
+}
+private fun createCodeFragment(
+    context: RsElement,
+    factory: (importTarget: RsItemsOwner) -> RsCodeFragment
 ): PsiCodeFragment {
     val freshMod = RsPsiFactory(context.project).createModItem(TMP_MOD_NAME, "")
```

Comment:
Can we use single `freshMod` for all code fragments (both `type` and `default value`) for all parameters? So that if some type was imported in `type` code fragment, user will not need to import it again in `default value`. E.g. consider some enum `Foo` with enum variant `Foo::Bar`, user enters `Foo` as new parameter type and `Foo::Bar` as this parameter default value

Reply:
That's a very good point! I'll change it. Also, is there a way for the temporary context module to "see" the same things as the refactored function? Importing works fine, but if I want to use some item that is in the same scope as the function, I currently still have to import it in the type/default value fragment.

Reply:
> Also, is there a way for the temporary context module to "see" the same things as the refactored function?

Maybe set `use path::to::function::containing::mod;` as fresh module text

Reply:
Hmm, it won't help with relative paths (like `super`, but it's not a big deal). I hoped that there would be some better way with setting the context of the module, but if not, this is a good enough fix :) Thanks.